integration-tests/tests/src/tests/mixed.ts

```diff
@@ -839,78 +2714,180 @@ describe("Mixed", () => {
+        expect(list[0]).equals("original");
+
+        const dictionary = objects[2].mixed;
+        expectRealmDictionary(dictionary);
+        expect(dictionary.key).equals("original");
       });
 
       it("throws when setting a list or dictionary outside a transaction", function (this: RealmContext) {
         const created = this.realm.write(() => {
-          return this.realm.create<IMixedSchema>(MixedSchema.name, { value: "original" });
+          return this.realm.create<IMixedSchema>(MixedSchema.name, { mixed: "original" });
         });
-        expect(created.value).equals("original");
-        expect(() => (created.value = ["a list item"])).to.throw(
+        expect(created.mixed).equals("original");
+        expect(() => (created.mixed = ["a list item"])).to.throw(
           "Cannot modify managed objects outside of a write transaction",
         );
-        expect(() => (created.value = { key: "a dictionary value" })).to.throw(
+        expect(() => (created.mixed = { key: "a dictionary value" })).to.throw(
           "Cannot modify managed objects outside of a write transaction",
         );
-        expect(created.value).equals("original");
+        expect(created.mixed).equals("original");
+      });
+
+      it("throws when setting a list item out of bounds", function (this: RealmContext) {
+        const { mixed: list } = this.realm.write(() => {
+          // Create an empty list as the Mixed value.
+          return this.realm.create<IMixedSchema>(MixedSchema.name, { mixed: [] });
+        });
+        expectRealmList(list);
+        expect(list.length).equals(0);
+
+        expect(() => {
+          this.realm.write(() => {
+            list[0] = "primitive";
+          });
+        }).to.throw("Requested index 0 calling set() on list 'MixedClass.mixed' when empty");
+
+        expect(() => {
+          this.realm.write(() => {
+            list[0] = [];
+          });
+        }).to.throw("Requested index 0 calling set_collection() on list 'MixedClass.mixed' when empty");
+
+        expect(() => {
+          this.realm.write(() => {
+            list[0] = {};
+          });
+        }).to.throw("Requested index 0 calling set_collection() on list 'MixedClass.mixed' when empty");
+      });
+
+      it("throws when setting a nested list item out of bounds", function (this: RealmContext) {
+        const { mixed: list } = this.realm.write(() => {
+          // Create a list containing an empty list as the Mixed value.
+          return this.realm.create<IMixedSchema>(MixedSchema.name, { mixed: [[]] });
+        });
+        expectRealmList(list);
+        expect(list.length).equals(1);
+
+        const nestedList = list[0];
+        expectRealmList(nestedList);
+        expect(nestedList.length).equals(0);
+
+        expect(() => {
+          this.realm.write(() => {
+            nestedList[0] = "primitive";
+          });
+        }).to.throw("Requested index 0 calling set() on list 'MixedClass.mixed' when empty");
```

Comment:
Shouldn't this be allowed?

Reply:
See [comment](https://github.com/realm/realm-js/pull/6513#discussion_r1532397116).